torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -159,10 +164,17 @@ class ProcessGroupNCCL : public ProcessGroup {
 
     // Store a reference to NCCL collective's outputs to be used by getFuture.
     std::shared_ptr<std::vector<at::Tensor>> outputs_;
+
     // Store streams that run FutureNCCL then callbacks.
     std::vector<std::shared_ptr<at::cuda::CUDAStream>>
         futureNCCLCallbackStreams_;
 
+    // Current rank of the node.
+    int rank_;
+
+    // Operation type that this work object refers to.
+    OpType opType_;
```

Comment:
Should we make `opType_` a member of the parent class `ProcessGroup::Work`? This kind of logging will be useful for other process groups as well